src/main/java/com/mraof/minestuck/client/gui/playerStats/EcheladderScreen.java

```diff
@@ -228,8 +276,14 @@ private List<Component> drawEffectIconsAndText(PoseStack poseStack, int currentR
 		mc.font.draw(poseStack, String.valueOf(ClientPlayerData.getBoondollars()), xOffset + 27 + mc.font.width("="), yOffset + 12, 0x0094FF);
 		//mc.fontRenderer.drawString("Rep: " + ClientPlayerData.getConsortReputation(), xOffset + 75 + mc.fontRenderer.getCharWidth('='), yOffset + 12, 0x0094FF);
 		
+		/**
+		 * here we are rendering the grist cache limit.
+		 * we show the current limit by calling the rungGrist array in grist helper.
+		 * this is seen on string.valueOf(GristHelper...
+		 */
 		mc.font.draw(poseStack, I18n.get(CACHE), xOffset + 24, yOffset + 138, 0x404040);
-		mc.font.draw(poseStack, "Unlimited", xOffset + 26, yOffset + 147, 0x0094FF);
+		mc.font.draw(poseStack, String.valueOf(GristHelper.rungGrist[currentRung]), xOffset + 26,
+				yOffset + 147, 0x0094FF);
```

Comment:
It's worth noting that this is the only change to this file that's directly related to the Grist Gutter, and it may be better to see if you can get the other changes into a different branch/PR.

Reply:
wuh